lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -3452,6 +3452,7 @@ ConformanceChecker::inferTypeWitnessesViaValueWitnesses(
       switch (reqt.getKind()) {
       case RequirementKind::Conformance:
       case RequirementKind::Superclass:
+        // FIXME: This is the wrong check
```

Comment:
We need to create a ConformsTo constraint for RequirementKind::Conformance, and a Subtype constraint for RequirementKind::Superclass.